DMCompiler/DM/Visitors/DMVisitorExpression.cs

```diff
@@ -192,6 +192,10 @@ public void VisitIdentifier(DMASTIdentifier identifier)
         public void VisitAssign(DMASTAssign assign) {
             var lhs = DMExpression.Create(_dmObject, _proc, assign.Expression, _inferredPath);
             var rhs = DMExpression.Create(_dmObject, _proc, assign.Value, lhs.Path);
+            if(lhs.TryAsConstant(out var _) || ((lhs.ValType & DMValueType.CompiletimeReadonly) == DMValueType.CompiletimeReadonly))
```

Comment:
This does not work. This is because `ValType` is actually a bit of an unfinished property and doesn't always reflect the type of its value.

`opendream_unimplemented` works by attaching the value type to the expression instead of the variable, and snowflaking it with dereference expressions which is less than ideal.